app/services/structured_fact_importer.rb

```diff
@@ -1,7 +1,7 @@
 class StructuredFactImporter < FactImporter
   def normalize(facts)
     # Remove empty values first, so nil facts added by flatten_composite imply compose
-    facts.keep_if { |k, v| v.present? }
+    facts.keep_if { |k, v| !v.nil? || (v.is_a?(String) && !v.empty?) }
```

Comment:
Maybe `delete_if { |k, v| v.nil? || (v.is_a?(String) && v.empty?) }` is easier to read